src/renderer/dx/DxRenderer.cpp

```diff
@@ -1196,6 +1497,51 @@ enum class CursorPaintType
     return S_OK;
 }
 
+// Routine Description:
+// - Paint terminal effects.
+// Arguments:
+// Return Value:
+// - S_OK or relevant DirectX error.
+[[nodiscard]] HRESULT DxEngine::PaintTerminalEffects() noexcept
+{
+    assert(_framebufferCapture && "You should have initialized this");
+
+    // Capture current frame in swap chain to a texture.
+    ::Microsoft::WRL::ComPtr<ID3D11Texture2D> swapBuffer;
+    RETURN_IF_FAILED(_dxgiSwapChain->GetBuffer(0, __uuidof(ID3D11Texture2D), (LPVOID*)&swapBuffer));

```

Comment:
I think there's a `IID_PPV_ARGS` macro that can replace the second and third params here in a cleaner fashion.